L1Trigger/Phase2L1ParticleFlow/plugins/L1CTJetFileWriter.cc

```diff
@@ -40,43 +42,82 @@ class L1CTJetFileWriter : public edm::one::EDAnalyzer<edm::one::SharedResources>
   void analyze(const edm::Event&, const edm::EventSetup&) override;
   void endJob() override;
   std::vector<ap_uint<64>> encodeJets(const std::vector<l1t::PFJet> jets);
+  std::vector<ap_uint<64>> encodeSums(const std::vector<l1t::EtSum> jets);
 
-  edm::EDGetTokenT<edm::View<l1t::PFJet>> jetsToken_;
   l1t::demo::BoardDataWriter fileWriterOutputToGT_;
+  std::vector<std::pair<edm::EDGetTokenT<edm::View<l1t::PFJet>>, edm::EDGetTokenT<edm::View<l1t::EtSum>>>> tokens_;
+  std::vector<std::pair<bool, bool>> tokensToWrite_;
+  //std::vector<edm::EDGetTokenT<edm::View<l1t::PFJet>>> jetsTokens_;
+  //std::vector<edm::EDGetTokenT<edm::View<l1t::EtSum>>> mhtTokens_;
 };
 
 L1CTJetFileWriter::L1CTJetFileWriter(const edm::ParameterSet& iConfig)
-    : nJets_(iConfig.getParameter<unsigned>("nJets")),
+    : collections_(iConfig.getUntrackedParameter<std::vector<edm::ParameterSet>>("collections",
+                                                                                 std::vector<edm::ParameterSet>())),
+      nJets_(iConfig.getParameter<unsigned>("nJets")),
       nFramesPerBX_(iConfig.getParameter<unsigned>("nFramesPerBX")),
       ctl2BoardTMUX_(iConfig.getParameter<unsigned>("TMUX")),
-      gapLengthOutput_(ctl2BoardTMUX_ * nFramesPerBX_ - 2 * nJets_),
+      gapLengthOutput_(ctl2BoardTMUX_ * nFramesPerBX_ - 2 * (nJets_ + 1) * collections_.size()),
       maxLinesPerFile_(iConfig.getParameter<unsigned>("maxLinesPerFile")),
       channelSpecsOutputToGT_{{{"jets", 0}, {{ctl2BoardTMUX_, gapLengthOutput_}, {0}}}},
-      jetsToken_(consumes<edm::View<l1t::PFJet>>(iConfig.getParameter<edm::InputTag>("jets"))),
       fileWriterOutputToGT_(l1t::demo::parseFileFormat(iConfig.getParameter<std::string>("format")),
                             iConfig.getParameter<std::string>("outputFilename"),
                             nFramesPerBX_,
                             ctl2BoardTMUX_,
                             maxLinesPerFile_,
-                            channelSpecsOutputToGT_) {}
+                            channelSpecsOutputToGT_) {
+  for (const auto& pset : collections_) {
+    edm::EDGetTokenT<edm::View<l1t::PFJet>> jetToken;
+    edm::EDGetTokenT<edm::View<l1t::EtSum>> mhtToken;
+    bool writeJetToken(false), writeMhtToken(false);
+    if (pset.exists("jets")) {
```

Comment:
Perhaps this could be `if (nJets_ > 0)`, safer against misconfigurations?

Reply:
Done, and with a configurable number of jets for each collection